zerver/lib/actions.py

```diff
@@ -2870,11 +2878,74 @@ def do_remove_default_stream(stream):
     DefaultStream.objects.filter(realm_id=realm_id, stream_id=stream_id).delete()
     notify_default_streams(realm_id)
 
+def do_create_default_stream_group(realm, group_name, streams):
+    # type: (Realm, Text, List[Stream]) -> None
+    default_streams = get_default_streams_for_realm(realm.id)
+    for stream in streams:
+        if stream in default_streams:
+            raise JsonableError(_("%s is a default stream and cannot be added to %s") % (stream.name, group_name))
+
+    try:
+        DefaultStreamGroup.objects.get(name=group_name, realm=realm)
+    except DefaultStreamGroup.DoesNotExist:
+        group = DefaultStreamGroup.objects.create(name=group_name, realm=realm)
+        group.streams = streams
+        group.save()
+        notify_default_stream_groups(realm)
+    else:
+        raise JsonableError(_("Default stream group %s already exists") % (group_name,))
```

Comment:
This is sorta an odd way to write this; I would use `DefaultStreamGroup.objects.get_or_create` and then do an `if not created` check; you'll repeat less code and the flow won't have the normal code path inside an exception handler.